platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/consensus/NonAncientEventWindow.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.consensus;
+
+import com.swirlds.common.context.PlatformContext;
+import com.swirlds.platform.event.GossipEvent;
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * Determines the window of rounds between the pending consensus round and the minimum round non-ancient. Provides the
+ * following information:
+ * <ol>
+ *     <li> latestConsensusRound - the latest round to have come to consensus.</li>
+ *     <li> minRoundNonAncient - the ancient threshold based on event birth round</li>
+ *     <li> minGenNonAncient - the ancient threshold based on event generation</li>
+ *     <li> pendingConsensusRound - the current round coming to consensus, i.e. 1  + the latestConsensusRound</li>
+ * </ol>
+ * <p>
+ * FUTURE WORK: Remove minGenNonAncient once we throw the switch to using minRoundNonAncient as the ancient threshold.
+ */
+public record NonAncientEventWindow(long latestConsensusRound, long minRoundNonAncient, long minGenNonAncient) {
+
+    /**
+     * @return the pending round coming to consensus, i.e. 1  + the latestConsensusRound
+     */
+    public long pendingConsensusRound() {
+        return latestConsensusRound + 1;
+    }
```

Comment:
We may not end up using this. But ok to leave until refactor is completed, easy to remove later if it goes unused.

Reply:
Until we are sending the diff of the `CurrentEffectiveRoster` which will indicate the `pendingConsensusRound` (Red Wire), we can use this (Green Wire) to communicate the `pendingConsensusRound` and set the proper `birthRound` in events.    I'll add a cleanup item to make sure that `pendingConsensusRound` is coming from the diff.  https://github.com/hashgraph/hedera-services/issues/10612